inference-engine/src/gna_plugin/transformations/split_convolution_with_large_buffer_size.cpp

```diff
@@ -28,9 +29,22 @@ static bool Convert(std::shared_ptr<ngraph::Node> conv,
     if (input_size <= GNALimitations::bufferMaxSize) {
         return false;
     }
-
-    uint32_t width = conv->get_input_shape(0).back();
-    uint32_t in_channels = conv->get_input_shape(0).at(1);
+    auto& input = conv->get_input_shape(0);
```

Comment:
it's better to move this check into the matcher for this transformation, as a predicate in the second argument of convolution wrap_type:
auto conv = ngraph::pattern::wrap_type<ngraph::opset7::Convolution>({ngraph::pattern::any_input(),
        ngraph::pattern::any_input()}, [](const ngraph::Output<ngraph::Node>& node) {
 { put convolution check here }
});

Reply:
OK